File: scrapers/add_news_to_db.js
import fs from 'fs';
import { fileURLToPath } from 'url';
import { dirname, join } from 'path';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

const newsPath = join(__dirname, 'data', 'news_items.json');
const dbPath = join(__dirname, 'db.json');

const newsData = JSON.parse(fs.readFileSync(newsPath, 'utf8'));
const dbData = JSON.parse(fs.readFileSync(dbPath, 'utf8'));

const newsWithIds = newsData.map((news, index) => {
  return {
    id: index + 1,
    title: news.title,
    category: news.category,
    link: news.link,
    image_url: news.image_url
  };
});

dbData.news = newsWithIds;

fs.writeFileSync(dbPath, JSON.stringify(dbData, null, 2), 'utf8');

console.log(`✅ Добавлено ${newsWithIds.length} новостей в db.json`);
console.log(`📊 Всего новостей: ${newsWithIds.length}`);

File: scrapers/add_news_to_db.py
import json
import os

script_dir = os.path.dirname(os.path.abspath(__file__))

news_path = os.path.join(script_dir, 'data', 'news_items.json')
db_path = os.path.join(script_dir, 'db.json')

print(f'Читаю новости из: {news_path}')
with open(news_path, 'r', encoding='utf-8') as f:
    news_data = json.load(f)

print(f'Читаю db.json из: {db_path}')
with open(db_path, 'r', encoding='utf-8') as f:
    db_data = json.load(f)

print('Обрабатываю новости...')
news_with_ids = []
for index, news in enumerate(news_data, start=1):
    news_with_ids.append({
        'id': index,
        'title': news.get('title', ''),
        'category': news.get('category', ''),
        'link': news.get('link', ''),
        'image_url': news.get('image_url')
    })

db_data['news'] = news_with_ids

print(f'Сохраняю {len(news_with_ids)} новостей в db.json...')
with open(db_path, 'w', encoding='utf-8') as f:
    json.dump(db_data, f, ensure_ascii=False, indent=2)

print(f'✅ Успешно добавлено {len(news_with_ids)} новостей в db.json')
print(f'📊 Всего новостей в базе: {len(news_with_ids)}')

